tool2schema/type_schema.py

```diff
@@ -215,11 +238,14 @@ def decode(self, value):
         # We have no way to know which type to decode to, so we try all of them
         # (ordered by priority) and return at the first one that produces a different value.
         for sub_type in self._sorted_sub_types():
-            if (dec := sub_type.decode(value)) != value:
+            if sub_type.validate(value) and (dec := sub_type.decode(value)) != value:
```

Comment:
Why?

Reply:
I thought it was more intuitive for the decode method to only be invoked with instances of the valid type, but it doesn't make any difference, I removed it with d3cd912